front/app/components/Form/index.tsx

```diff
@@ -266,18 +271,22 @@ const Form = memo(
             </FormContext.Provider>
           </APIErrorsContext.Provider>
         </Box>
-        {layoutType === 'fullpage' ? (
-          <ButtonBar
-            onSubmit={handleSubmit}
-            apiErrors={Boolean(
-              apiErrors?.values?.length && apiErrors?.values?.length > 0
+        {showSubmitButton && (
+          <>
+            {layoutType === 'fullpage' ? (
+              <ButtonBar
+                onSubmit={handleSubmit}
+                apiErrors={Boolean(
+                  apiErrors?.values?.length && apiErrors?.values?.length > 0
+                )}
+                processing={loading}
+              />
+            ) : submitOnEvent ? (
+              <InvisibleSubmitButton onClick={handleSubmit} />
+            ) : (
+              <Button onClick={handleSubmit}>Button</Button>
```

Comment:
Is the "Button" text within the Button supposed to be here?

Reply:
Seemed odd for me as-well but I didn't want to change some existing work without proper direction or making breaking changes out of scope